### PR TITLE
feat(schemas,core,console): add SAML application type

### PR DESCRIPTION
**Pull request by darcyYe**
## Summary

add `SAML` application type, with following changes:
1. add `SAML` to enum application type
2. can not create/update/delete SAML apps with `POST /applications` and `PATCH/DEL /applications/:id` APIs
3. update the type in @logto/console as well to prevent build errors resulted by adding possible `SAML` value to `application.type` type.

## Testing

Should be covered by integration tests.

## Checklist

- [ ] `.changeset`
- [ ] unit tests
- [x] integration tests
- [ ] necessary TSDoc comments

### CODE REVIEW

File: packages/core/src/routes/applications/application.ts [hunk 1]
```diff
@@ -147,10 +147,14 @@ export default function applicationRoutes<T extends ManagementApiRouter>(
...
       status: [200, 400, 422, 500],
     }),
-
+    // eslint-disable-next-line complexity
     async (ctx, next) => {
       const { oidcClientMetadata, protectedAppMetadata, ...rest } = ctx.guard.body;
 
+      if (rest.type === ApplicationType.SAML) {
+        throw new RequestError('application.use_saml_app_api');
+      }
```
Comment (simeng-li) [inline packages/core/src/routes/applications/application.ts:156]: Should omit the `SAML application type from `applicationCreateGuard`, to it won't be available on the API docs as well.

File: packages/core/src/routes/applications/application.ts [hunk 2]
```diff
@@ -262,6 +266,11 @@ export default function applicationRoutes<T extends ManagementApiRouter>(
 
       const { isAdmin, protectedAppMetadata, ...rest } = body;
 
+      const pendingUpdateApplication = await queries.applications.findApplicationById(id);
+      if (pendingUpdateApplication.type === ApplicationType.SAML) {
+        throw new RequestError('application.use_saml_app_api');
+      }
```
Comment (simeng-li) [inline packages/core/src/routes/applications/application.ts:272]: How about `saml_app_read_only`?

Comment (darcyYe) [reply packages/core/src/routes/applications/application.ts:279]: Based on the offline discussions, we are now disabling the `GET /applications(/:id)?` API for SAML apps. When we need to modify the console, we will then allow the `GET /applications/(:id)?` API, and at that time, we can change it to `saml_app_read_only`.

File: packages/console/src/assets/docs/guides/types.ts [hunk 3]
```diff
@@ -1,7 +1,8 @@
-import { type ApplicationType } from '@logto/schemas';
 import { type MDXProps } from 'mdx/types';
 import { type LazyExoticComponent, type ComponentType, type SVGProps } from 'react';
 
+import { type ApplicationTypeWithoutSaml } from '@/types/applications';
```
Comment (gao-sun) [inline packages/console/src/assets/docs/guides/types.ts:4]: why do we need `ApplicationTypeWithoutSaml`?

File: packages/console/src/pages/EnterpriseSsoDetails/IdpInitiatedAuth/ConfigForm.tsx [hunk 4]
```diff
@@ -177,17 +177,29 @@ function ConfigForm({
...
+                        (
+                          application
+                        ): application is Omit<Application, 'type'> & {
+                          type: Exclude<ApplicationType, ApplicationType.SAML>;
+                        } =>
+                          /**
+                           * See definition of `applicationsSearchUrl`, there is only non-third party SPA/Traditional applications here, and SAML applications are always third party secured by DB schema, we need to manually exclude other application types here to make TypeScript happy.
+                           */
+                          application.type !== ApplicationType.SAML
+                      )
```
Comment (simeng-li) [inline packages/console/src/pages/EnterpriseSsoDetails/IdpInitiatedAuth/ConfigForm.tsx:191]: nit: use a useMemo? Too many inline loops here.

**Review by simeng-li — APPROVED** [review]
LGTM

**Review by charIeszhao — APPROVED** [review]

**Pull request merged**